kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyConfigSecret.java

```diff
@@ -43,33 +62,56 @@ static String secretName(KafkaProxy primary) {
     @Override
     protected Secret desired(KafkaProxy primary,
                              Context<KafkaProxy> context) {
-
+        // formatter=off
         return new SecretBuilder()
                 .editOrNewMetadata()
-
-                .withName(secretName(primary))
-                .withNamespace(primary.getMetadata().getNamespace())
+                    .withName(secretName(primary))
+                    .withNamespace(primary.getMetadata().getNamespace())
                 .endMetadata()
-                .withStringData(Map.of(CONFIG_YAML_KEY,
-                        """
-                                adminHttp:
-                                  endpoints:
-                                    prometheus: {}
-                                virtualClusters:
-                                  demo:
-                                    targetCluster:
-                                      bootstrap_servers: TARGET_BOOTSTRAP_SERVERS
-                                    clusterNetworkAddressConfigProvider:
-                                      type: PortPerBrokerClusterNetworkAddressConfigProvider
-                                      config:
-                                        bootstrapAddress: localhost:9292
-                                        brokerAddressPattern: BROKER_ADDRESS_PATTERN
-                                    logNetwork: false
-                                    logFrames: false
-                                """
-                                .replace(
-                                        "TARGET_BOOTSTRAP_SERVERS", primary.getSpec().getBootstrapServers())
-                                .replace("BROKER_ADDRESS_PATTERN", ProxyService.serviceName(primary))))
+                .withStringData(Map.of(CONFIG_YAML_KEY, generateProxyConfig(primary)))
                 .build();
     }
+
+    String generateProxyConfig(KafkaProxy primary) {
```

Comment:
nit: why `primary` ?

Reply:
That's what JOSDK invariably calls it, and so I've stuck with that for now. I'm happy to change it, but perhaps better in a dedicated PR?

Reply:
I have no desire to fight the defaults.